README.md

```diff
@@ -4,222 +4,1008 @@
+
+JCLI has special support for this type, as it is used to mark an argument as optional. This type is publicly imported by JCLI, so you don't have to import anything extra.
+
+Anyway, all we want to do is make our `mode` argument a `Nullable`, so JCLI knows it's optional:
 
-You can add the [JIOC](https://code.dlang.org/packages/jioc) library into your project, and then create and pass a `ServiceProvider` into the constructor
-of `CommandLineInterface`. Afterwards, all command objects are created with dependency injection via JIOC's `Injector.construct`.
+```d
+@Command(null, "The default command.")
+struct DefaultCommand
+{
+    @CommandPositionalArg(0, "number", "The number to check.")
+    int number;
+
+    @CommandNamedArg("mode", "Which mode to use.")
+    Nullable!Mode mode;
 
-This is the *only* way `CommandLineInterface` is able to pass data into a command's constructor.
+    int onExecute()
+    {
+        if(this.mode.get(Mode.normal) == Mode.normal)
+            return number % 2 == 0 ? 1 : 0;
+        else
+            return number % 2;
+    }
+}
+```
 
-Please see [this](https://github.com/SealabJaster/jcli/tree/master/examples/05-dependency-injection) example for more information.
+The first change is that `Mode mode;` has now become `Nullable!Mode mode`, to make it, quite literally, a `Mode` that is `Nullable`.
 
-## Help text
+The other change we made is that inside of `onExecute` we now use `this.mode.get(Mode.normal)`.
 
-`CommandLineInterface` will automatically generate help text for any given command, and if no command is specified (or found) then it'll show a list of all commands, or commands
-similar to the arguments given to it.
+The `Nullable.get` function will either return us the value stored in the `Nullable`, or if the `Nullable` is null it will return to us the value we pass to it.
 
-Help text is shown either by using an unknown command, or passing either `-h` or `--help`. This does mean however that commands cannot use these as argument names.
+So by doing `get(Mode.normal)` we're saying "Give us the value the user passed in. Or, if the user didn't pass in a value, default to `Mode.normal`".
 
-Here is an example of the help text for a specific command:
+First, let's look at the help text, as it very slightly changes for nullable arguments:
 
-```text
-> .\aim.exe secrets set -h
-Usage: secrets set {0/name} {1/value} <[v|verbose]>
+```bash
+$> ./mytool --help
+Usage: DEFAULT {0/number} <[mode]>
 
 Description:
-    Sets the value of a secret.
+    The default command.
 
 Positional Args:
-    0,name                       - The name of the secret to set the value of.
-    1,value                      - The value to give the secret.
+    0,number                     - The number to check.
 
 Named Args:
-    -v,--verbose                 - Show verbose output.
+    --mode                       - Which mode to use.
+```
+
+Notice the "Usage:" line. The `[mode]` has now become `<[mode]>` to indicate it is optional.
```

Comment:
As I said earlier, `[mode]` (i.e. square brackets) means it's optional (see [mkdir](https://man7.org/linux/man-pages/man1/mkdir.1.html) for example).